cypress/integration/products/productsVariants.js

```diff
@@ -0,0 +1,191 @@
+    channelsUtils.deleteChannels(startsWith);
+
+    const name = `${startsWith}${faker.random.number()}`;
+    channelsUtils
+      .getDefaultChannel()
+      .then(channel => {
+        defaultChannel = channel;
+        cy.fixture("addresses");
+      })
+      .then(fixtureAddresses =>
+        shippingUtils.createShipping({
+          channelId: defaultChannel.id,
+          name,
+          address: fixtureAddresses.plAddress
+        })
+      )
+      .then(() => (warehouse = shippingUtils.getWarehouse()));
+
+    productUtils
+      .createTypeAttributeAndCategoryForProduct(name, attributeValues)
+      .then(() => {
+        attribute = productUtils.getAttribute();
+        productType = productUtils.getProductType();
+        category = productUtils.getCategory();
+      });
+  });
+
+  beforeEach(() => {
+    cy.clearSessionData().loginUserViaRequest();
+  });
+
+  it("should create variant visible on frontend", () => {
+    const name = `${startsWith}${faker.random.number()}`;
+    const price = 10;
+    let createdProduct;
+
+    product
+      .createProduct(attribute.id, name, productType.id, category.id)
+      .then(resp => {
+        createdProduct = resp.body.data.productCreate.product;
+        product.updateChannelInProduct({
+          productId: createdProduct.id,
+          channelId: defaultChannel.id
+        });
+        cy.visit(`${urlList.products}${createdProduct.id}`);
+        variantsSteps.createFirstVariant({
+          sku: name,
+          warehouseId: warehouse.id,
+          price,
+          attribute: attributeValues[0]
+        });
+        getProductVariants(createdProduct.id, defaultChannel.slug);
+      })
+      .then(variants => {
+        expect(variants[0]).to.have.property("name", attributeValues[0]);
+        expect(variants[0].pricing.price.gross).to.have.property(
+          "amount",
+          price
+        );
+      });
+  });
+  it("should create several variants", () => {
+    const name = `${startsWith}${faker.random.number()}`;
+    const secondVariantSku = `${startsWith}${faker.random.number()}`;
+    const variants = [{ amount: 7 }, { name: attributeValues[1], amount: 16 }];
+    let createdProduct;
+
+    productUtils
+      .createProductInChannel({
+        name,
+        attributeId: attribute.id,
+        channelId: defaultChannel.id,
+        warehouseId: warehouse.id,
+        productTypeId: productType.id,
+        categoryId: category.id,
+        price: variants[0].amount
+      })
+      .then(() => {
+        createdProduct = productUtils.getCreatedProduct();
+        cy.visit(`${urlList.products}${createdProduct.id}`);
+        variantsSteps.createVariant({
+          sku: secondVariantSku,
+          warehouseName: warehouse.name,
+          attributeName: variants[1].name,
+          price: variants[1].amount
+        });
+      })
+      .then(() => getProductVariants(createdProduct.id, defaultChannel.slug))
+      .then(variantsResp => {
```

Comment:
We could destructure the response like
```suggestion
      .then(([namedProp, otherNamedProp, anotherNamedProp]) => {
```